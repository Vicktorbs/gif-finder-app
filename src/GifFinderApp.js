import React, { useState } from 'react';
import AddCategory from './components/AddCategory'
import { GifGrid } from './components/GifGrid';

const GifFinderApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)
    
    return (
        <>
            <h2>GifFinderApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr/>
            <ol>
                { categories.map(category => 
                    <GifGrid 
                        key={ category }
                        category={ category }
                    />
                ) }
            </ol>
        </>
    )
}
export default GifFinderApp;