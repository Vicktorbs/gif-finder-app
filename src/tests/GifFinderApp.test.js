import { shallow } from "enzyme"
import GifFinderApp from "../GifFinderApp"

describe('Test on file GifFinderApp', () => {
    
    test('should show <GifFinderApp /> correctly', () => {
        const wrapper = shallow(<GifFinderApp />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show a category list', () => {
        const categories = ['Hinata', 'Chitanda'];
        const wrapper = shallow(<GifFinderApp defaultCategories={ categories } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})
