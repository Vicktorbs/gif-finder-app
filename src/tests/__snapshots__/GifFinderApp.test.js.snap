// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on file GifFinderApp should show <GifFinderApp /> correctly 1`] = `
<Fragment>
  <h2>
    GifFinderApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test on file GifFinderApp should show a category list 1`] = `
<Fragment>
  <h2>
    GifFinderApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hinata"
      key="Hinata"
    />
    <GifGrid
      category="Chitanda"
      key="Chitanda"
    />
  </ol>
</Fragment>
`;
